### ISSUE TITLE
Reflect.callMethod() crashes on Java when defaulting arguments

### ISSUE THREAD

**Issue by ckentgeorge** [HaxeFoundation/haxe#8975]
Running Haxe 4.0.2 and hxjava 3.2.0, the following code crashes when attempting `Reflect.callMethod()` with `Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 1`.

Here's the a simplified example that crashes:
```
class Remote
{
   public function new() {}

   public function doIt
   (
      prefix:String, 
      suffix:String = "bet"
   )
   {
      return prefix + suffix;
   }
}

#if !macro
class Main
{
   private static function main():Void
   {
      var remote = new Remote();

      // This works.
      trace(remote.doIt("Alpha"));
      
      // This doesn't when targetting Java.
      trace(
         Reflect.callMethod(
            remote,
            remote.doIt,
            ["Alpha"]
         )
      );
   }
}
#end
```

**Comment by RealyUniqueName** [comment]
I think similar remark should be added to `Reflect.callMethod` doc (from `Type.createInstance()`):
```
if the number of elements in args does not match the expected number of constructor arguments, or if any argument has an invalid type, or if cl has no own constructor, the result is unspecified.

In particular, default values of constructor arguments are not guaranteed to be taken into account.
```

**Comment by ncannasse** [comment]
Regarding optional arguments / default values support, which platforms are concerned with this limitation ? If it's only Java maybe something should be done for this platform.

**Comment by ckentgeorge** [comment]
This worked just fine in Haxe 3. We're in a tough spot if support for this is intentionally dropped for Haxe 4. For example, Haxe script becomes fairly less usable when targeting Java.

**Comment by RealyUniqueName** [comment]
If it was working in Haxe 3, then it's a regression of Haxe 4 and should be fixed in 4.0.3

**Comment by RealyUniqueName** [comment]
Waiting for CI before merging into 4.0_bugfix branch
